src/misc/sw-template.js

```diff
@@ -278,6 +279,38 @@ function WebpackServiceWorker(params) {
     // * urlString -- url to match from the CACHE_NAME
     // * event.request -- original Request to perform fetch() if necessary
 
+    if (responseStrategy === "network-first") {
+      event.respondWith(
+        fetch(event.request)
+        .then((response) => {
+          if (DEBUG) {
+            console.log('[SW]:', `URL [${ urlString }] from network`);
+          }
+
+          // always fresh cache response for future usage
+          caches.open(CACHE_NAME).then((cache) => {
+            cache.put(event.request, response.clone());
+            if (DEBUG) {
+              console.log('[SW]:', 'Cache asset: ' + urlString);
+            }
+          })
+
+          return response
+        })
+        .catch(() => {
+          if (DEBUG) {
+            console.log('[SW]:', `URL [${ urlString }] from cache if possible`);
+          }
+
+          return caches.match(event.request);
```

Comment:
Also if matching is caches fails, you must check if it's a navigation request and if so, try to load navigation fallback. As here: https://github.com/NekR/offline-plugin/blob/v4/src/misc/sw-template.js#L319